data/deduplicator/truncate_test.go

```diff
@@ -47,20 +48,20 @@ var _ = Describe("Truncate", func() {
 
 			Context("with deduplicator", func() {
 				BeforeEach(func() {
-					testDataset.Deduplicator = &data.DeduplicatorDescriptor{}
+					testDataset.Deduplicator = &data.DeduplicatorDescriptor{Name: "truncate"}
 				})
 
-				It("returns false if the deduplicator name is missing", func() {
+				It("returns false if the deduplicator name is empty", func() {
+					testDataset.Deduplicator.Name = ""
 					Expect(testFactory.CanDeduplicateDataset(testDataset)).To(Equal(false))
 				})
 
-				It("returns true if the deduplicator name is not truncate", func() {
+				It("returns true if the deduplicator name does not match", func() {
```

Comment:
Right you are. I'm going to make the changes on this PR https://github.com/tidepool-org/platform/pull/104, though, since the same problem exists on the hash deduplicator tests.